test/integration/publishing/send-updates-organisation.test.js

```diff
@@ -0,0 +1,202 @@
+      const unpublishedBefore = await db.organisation.findAll({ where: { published: null } })
+
+      await publish()
+
+      const unpublishedAfter = await db.organisation.findAll({ where: { published: null } })
+      expect(unpublishedBefore).toHaveLength(numberOfRecords)
+      expect(unpublishedAfter).toHaveLength(0)
+    })
+
+    test('should process all records when there are equal number of records than publishingConfig.dataPublishingMaxBatchSizePerDataSource', async () => {
+      const numberOfRecords = publishingConfig.dataPublishingMaxBatchSizePerDataSource
+      await db.organisation.bulkCreate([...Array(numberOfRecords).keys()].map(x => { return { ...mockOrganisation1, sbi: mockOrganisation1.sbi + x } }))
+      const unpublishedBefore = await db.organisation.findAll({ where: { published: null } })
+
+      await publish()
+
+      const unpublishedAfter = await db.organisation.findAll({ where: { published: null } })
+      expect(unpublishedBefore).toHaveLength(numberOfRecords)
+      expect(unpublishedAfter).toHaveLength(0)
+    })
+
+    test('should process publishingConfig.dataPublishingMaxBatchSizePerDataSource records when there are more records than publishingConfig.dataPublishingMaxBatchSizePerDataSource', async () => {
+      const numberOfRecords = 1 + publishingConfig.dataPublishingMaxBatchSizePerDataSource
+      await db.organisation.bulkCreate([...Array(numberOfRecords).keys()].map(x => { return { ...mockOrganisation1, sbi: mockOrganisation1.sbi + x } }))
+      const unpublishedBefore = await db.organisation.findAll({ where: { published: null } })
+
+      await publish()
+
+      const unpublishedAfter = await db.organisation.findAll({ where: { published: null } })
+      expect(unpublishedBefore).toHaveLength(numberOfRecords)
+      expect(unpublishedAfter).toHaveLength(numberOfRecords - publishingConfig.dataPublishingMaxBatchSizePerDataSource)
+    })
+
+    test('should process all records after the second publish when there are more records than publishingConfig.dataPublishingMaxBatchSizePerDataSource', async () => {
+      const numberOfRecords = 1 + publishingConfig.dataPublishingMaxBatchSizePerDataSource
+      await db.organisation.bulkCreate([...Array(numberOfRecords).keys()].map(x => { return { ...mockOrganisation1, sbi: mockOrganisation1.sbi + x } }))
+      const unpublishedBefore = await db.organisation.findAll({ where: { published: null } })
+
+      await publish()
+      const unpublishedAfterFirstPublish = await db.organisation.findAll({ where: { published: null } })
+
+      await publish()
+      const unpublishedAfterSecondPublish = await db.organisation.findAll({ where: { published: null } })
+
+      expect(unpublishedBefore).toHaveLength(numberOfRecords)
+      expect(unpublishedAfterFirstPublish).toHaveLength(numberOfRecords - publishingConfig.dataPublishingMaxBatchSizePerDataSource)
+      expect(unpublishedAfterSecondPublish).toHaveLength(0)
+    })
+  })
+})
```

Comment:
Can we add test for 2 or more concurrent db.organisation.findAll() processes ?

Reply:
How do you propose that can be tested?